parsec/core/gui/central_widget.py

```diff
@@ -199,10 +230,10 @@ def handle_event(self, event, **kwargs):
                 ),
             )
         elif event == CoreEvent.SHARING_UPDATED:
-            new_entry = kwargs["new_entry"]
-            previous_entry = kwargs["previous_entry"]
-            new_role = getattr(new_entry, "role", None)
-            previous_role = getattr(previous_entry, "role", None)
+            new_entry: WorkspaceEntry = kwargs["new_entry"]  # type: ignore
+            previous_entry: Optional[WorkspaceEntry] = kwargs["previous_entry"]  # type: ignore
+            new_role = new_entry.role
+            previous_role = previous_entry.role if previous_entry is not None else None
```

Comment:
Are we sure that if `new_entry` and `previous_entry` are not None, they will always have a `role` attribute? The use of `getattr` here makes me think that it's not always the case.

Reply:
`new_entry` and `previous_entry` are of type `WorkspaceEntry` which has a `role` attribute, so we are sure of this ;-)

I think the use of `getattr(previous_entry, "role", None)` was to simply handle the case when `previous_entry` is `None` (and then the same pattern was applied for `new_entry` for consistency)